### PR TITLE
cleanup-webparsing: Clean up web parsing

### PR DESCRIPTION
**Pull request by Ian-Stewart-Binks**
This PR cleans up a lot of the WebParsing module.

### CODE REVIEW

File: hs/WebParsing/TimeTableParser.hs [hunk 1]
```diff
@@ -184,21 +191,3 @@ processCourseTable course = do
         containsTut sesh = any (maybe False (T.isPrefixOf "T") . tutorialSection) $ tutorials sesh
         containsPrac sesh = any (maybe False (T.isPrefixOf "P") . tutorialSection) $ tutorials sesh
 
-main :: IO ()
-main = do
-    rsp <- simpleHTTP (getRequest $ timetableUrl ++ "sponsors.htm")
-    body <- getResponseBody rsp
-    let _ = getDeptList $ parseTags body
```
Comment (Ian-Stewart-Binks) [inline hs/WebParsing/TimeTableParser.hs]: Wasn't sure if this function is ever used. Removed it, it seems pretty pointless.

File: hs/WebParsing/TimeTableParser.hs [hunk 2]
```diff
@@ -27,35 +36,32 @@ data CourseSlot =
...
+-- used to deal with corner case found in "csc.html"
 expandNote :: [T.Text] -> [T.Text]
-expandNote row =  if ( (T.take 4 (head row)) == "NOTE" )
-                  then ["", "", ""] ++ (tail row)
-                  else row
+expandNote row
+    | T.take 4 (head row) == "NOTE" = ["", "", ""] ++ tail row
+    | otherwise = row
 
--- | takes in a department pagversace versaceversace versacee name, extracts the html table, partitions into a list of all
```
Comment (Ian-Stewart-Binks) [inline hs/WebParsing/TimeTableParser.hs]: @shaftoes versace?

File: hs/WebParsing/ParsingHelp.hs [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {-# LANGUAGE OverloadedStrings #-}
 module WebParsing.ParsingHelp
    (CoursePart,
-    (-:),
     (~:),
+    (-:),
```
Comment (Ian-Stewart-Binks) [inline hs/WebParsing/ParsingHelp.hs]: I'm not quite sure why these operators are necessary. They seem to just obfuscate function application.

**Comment by david-yz-liu** [comment]
Thanks, @Ian-Stewart-Binks! Many improvements here.

File: hs/WebParsing/ParsingHelp.hs (hunk 3 above)
Comment (david-yz-liu) [inline hs/WebParsing/ParsingHelp.hs]: I would tend to agree. The few locations where this is used can probably be re-written.

**Pull request merged**